test/test-lex.cpp

```diff
@@ -113,6 +113,62 @@ TEST_F(test_lex, lex_block_comments) {
   }
 }
 
+TEST_F(test_lex, lex_unopened_block_comment) {
+  {
+    error_collector v;
+    padded_string input(u8"hello */"_sv);
+    lexer l(&input, &v);
+    l.skip();  // identifier
+    l.skip();  // star
+    l.skip();  // slash
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
+                              error_unopened_block_comment, comment_close,
+                              offsets_matcher(&input, 6, u8"/"))));
```

Comment:
Must fix: The error should span the entire `*/`, not just the `*`.

Must fix: Your test says `/` but really only the `*` is highlighted. Write the correct character. (`offsets_matcher` only checks the string length, not its contents. I should fix that...)

Nit: Use strlen to improve test readability.

```suggestion
                              offsets_matcher(&input, strlen(u8"hello "), u8"*/"))));
```